### PR TITLE
Add Payment Management System 

### PR DESCRIPTION
**Pull request by ChengguiSun**
Payment management system is added. Detailed information about this PR/payment system are listed as below:

- plan options: monthly plan, quartely plan, yearly plan;
- payment method include: UnionPay, WeChat Pay, Alipay;
- quota management: reguar membership and professional membership.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

  

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
##

##

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: api/server/routes/payments.js [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+const Stripe = require('stripe');
+
+// Convert callback-based functions to promises
+const baseUrl = process.env.BASE_URL;
+const stripe = Stripe(process.env.STRIPE_SECRET_KEY);
+
+// planPricing
+const planPricing = {
+  month: {
+    price: '120.00',
```
Comment (wtzju) [inline api/server/routes/payments.js]: Is that possible to change the price from a string to a number?

File: client/src/components/Subscription/index.tsx [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+  const navigate = useNavigate();
+  const [subscriptionDueDate, setSubscriptionDueDate] = useState('');
+  const [subscribedPlanId, setSubscribedPlanId] = useState('');
+
+  const getUserByIdQuery = useGetUserByIdQuery(userId);
+
+  const subscriptionOptions = [
+    {
+      planId: 'month',
+      price: '120.00',
```
Comment (wtzju) [inline client/src/components/Subscription/index.tsx:28]: Is that possible to change the price from a string to a number here too?

File: package-lock.json [hunk 3]
```diff
@@ -7706,10 +7722,9 @@
       "license": "MIT"
     },
     "node_modules/@types/react": {
-      "version": "18.2.11",
-      "resolved": "https://registry.npmjs.org/@types/react/-/react-18.2.11.tgz",
-      "integrity": "sha512-+hsJr9hmwyDecSMQAmX7drgbDpyE+EgSF6t7+5QEBAn1tQK7kl1vWZ4iRf6SjQ8lk7dyEULxUmZOIpN0W5baZA==",
-      "license": "MIT",
```
Comment (wtzju) [inline package-lock.json:7712]: I am just curious about the license removal. What is the new license type?

**Review by wtzju — APPROVED** [review]
I feel the "price" should be changed from a string to a number. Other than this, the PR looks good to me.

File: package-lock.json (hunk 3 above)
Comment (ChengguiSun) [reply package-lock.json:7712]: > I am just curious about the license removal. What is the new license type?

I didn't notice this. Probably this is an automatic update when I install or uninstall some packaage

File: client/src/components/Subscription/index.tsx (hunk 2 above)
Comment (ChengguiSun) [reply client/src/components/Subscription/index.tsx:28]: > Is that possible to change the price from a string to a number here too?

updated

**Comment by ChengguiSun** [comment]
> I feel the "price" should be changed from a string to a number. Other than this, the PR looks good to me.

thanks for your review. I updated this.

**Pull request merged**